docs/spec/enums.rst

```diff
@@ -0,0 +1,342 @@
+the most common of these rules. The lesser-used rules are optional. Some
+of these rules may be impossible to evaluate and enforce statically in cases 
+where dynamic values are used.
+
+* If an attribute is defined in the class body with a type annotation but
+  with no assigned value, a type checker should assume this is a non-member
+  attribute::
+
+    class Pet(Enum):
+        genus: str  # Non-member attribute
+        species: str  # Non-member attribute
+
+        CAT = 1  # Member attribute
+        DOG = 2  # Member attribute
+
+  Within a type stub, members can be defined using the actual runtime values,
+  or a placeholder of ``...`` can be used::
+
+    class Pet(Enum):
+        genus: str  # Non-member attribute
+        species: str  # Non-member attribute
+
+        CAT = ...  # Member attribute
+        DOG = ...  # Member attribute
+
+* Members defined within an enum class should not include explicit type
+  annotations. Type checkers should infer a literal type for all members.
+  A type checker should report an error if a type annotation is used
+  for an enum member because this type will be incorrect and misleading
+  to readers of the code::
+
+    class Pet(Enum):
+        CAT = 1  # OK
+        DOG: int = 2  # Type checker error
+
+    reveal_type(Pet.CAT)  # Revealed type is Literal[Pet.CAT]
+    reveal_type(Pet.DOG)  # Revealed type is Literal[Pet.DOG]
+
+* Methods, callables, and descriptors (including properties) that are defined
+  in the class are not treated as enum members by the ``EnumType`` metaclass
+  and should likewise not be treated as enum members by a type checker::
+
+    def identity(__x): return __x
+
+    class Pet(Enum):
+        CAT = 1  # Member attribute
+        DOG = 2  # Member attribute
+        
+        converter = lambda __x: str(__x)  # Non-member attribute
+        transform = identity  # Non-member attribute
+
+        @property
+        def species(self) -> str:  # Non-member property
+            return "mammal"
+        
+        def speak(self) -> None:  # Non-member method
+            print("meow" if self is Pet.CAT else "woof")
+
+
+* If using Python 3.11 or newer, the ``enum.member`` and ``enum.nonmember``
+  classes can be used to unambiguously distinguish members from non-members.
+  Type checkers should support these classes::
+
+    class Example(Enum):
+        a = member(1)  # Member attribute
+        b = nonmember(2)  # Non-member attribute
+
+        @member
+        def c(self) -> None:  # Member method
+            pass
+
+    reveal_type(Example.a)  # Revealed type is Literal[Example.a]
+    reveal_type(Example.b)  # Revealed type is int
+    reveal_type(Example.c)  # Revealed type is Literal[Example.c]
+
+
+* An enum class can define a class symbol named ``_ignore_``. This can be a list
+  of names or a string containing a space-delimited list of names that are
+  excluded from the list of members at runtime. Type checkers may support this
+  mechanism::
```

Comment:
```suggestion
* An enum class can define a class symbol named ``_ignore_``. This can be a list
  of names or a string containing a space-delimited list of names that are
  removed from the enum class at runtime. Type checkers may support this
  mechanism::
```